### ISSUE TITLE
[NEWS] Nowe zasoby - grupowa edycja ceny i liczby przedmiotów 

### ISSUE THREAD

**Issue by MartaNowaczyk** [allegro/allegro-api#924]
Udostępniliśmy zasoby, które pozwolą wam:

- **zmienić cenę** w wielu ofertach jednocześnie,
```
PUT /sale/offer-price-change-commands/{commandId} 
GET /sale/offer-price-change-commands/{commandId} 
GET /sale/offer-price-change-commands/{commandId}/tasks
``` 
- **zmienić liczbę przedmiotów** w wielu ofertach jednocześnie 
```
PUT /sale/offer-quantity-change-commands/{commandId} 
GET /sale/offer-quantity-change-commands/{commandId}
GET /sale/offer-quantity-change-commands/{commandId}/tasks
``` 

Więcej znajdziecie w naszym [poradniku](https://developer.allegro.pl/my_offers/#3).

**Comment by MartaNowaczyk** [comment]
@dostawcaopon o ile dobrze cię rozumiem, to nie da się zrobić tego w ten sposób. W każdym requeście podajesz tylko tylko raz cenę lub liczbę przedmiotów i zmienisz ją dla wybranych ofert. Czyli np. w wielu ofertach zmiansz cenę na 100 zł.

**Comment by mpysz** [comment]
To o czym pisze @dostawcaopon byłoby bardzo przydatne. Jeśli ktoś ma wystawione kilka tysięcy ofert i prowadzi sprzedaż za pomocą różnych kanałów (nie tylko Allegro) to ilość sztuk na magazynie szybko się zmienia i konieczne jest częste aktualizowanie stanów przez API Allegro, żeby uniknąć sprzedania produktów, których już nie ma.
Zamiast robić 500 requestów co 10 minut szybciej byłoby zrobić 1 request, a w nim podać:
`
"offers": [
    {
        "id":"7660573029",
        "stock": 12
    },
    {
        "id":"7644576839",
        "stock": 3
    }
]
`

**Comment by dostawcaopon** [comment]
dokładnie, bardzo by sie przydało, przestalibyśmy bombardować wasz serwer requestami zmiany cen i ilości. Przy naszej ilości ofert (kilkadziesiąt tysięcy) tych żądań jest naprawdę dużo.

**Comment by PawelTaberski** [comment]
Jednakże mniejszym obciążeniem jest dla nas otrzymanie większej ilości małych requestów, niż kilku dużych i niweluje to ilość problemów i błędów, które bardzo często zdarzają się w dużych requestach. Dlatego raczej nie wdrożymy możliwości bulkowej zmiany wielu ofert z różną daną.

**Comment by wwwfactory** [comment]
Dobry wieczór, podłączamy się do wątku, 
reasumując,
jeżeli mam 1000 towarów i każdy ma inną ilość to 
PUT /sale/offer-quantity-change-commands/{commandId}  wywołujemy 1000 razy,
tak samo z cenami?

i troszkę odejście od tematu,
może warto rozważyć:
PUT /sale/offer-quantity-change-commands/{commandId}  z ilością === 0 jest równoznaczne z zakończeniem aukcji (put w status END) 

Pozdrawiamy wwwfactory

**Comment by wwwfactory** [comment]
Witamy ponownie,
pojawiło się jeszcze jedno pytanie, 
wywołuję:
PUT /sale/offer-quantity-change-commands/{commandId}
dla  offerId = xxxxxxxxxxxxxx
jak będzie wyglądał status tej oferty w trakcie zmiany ilości towaru (lub zmiany ceny) do momentu zakończenia przez wasze serwery procesowania nad tym zleceniem ? 
Jeżeli oferta była w statusie ACTIVE to zmiani się na NEW a następnie na ACTIVE czy status zawsze będzie ACTIVE ?
analogicznie dla END/ENDED w przypadku zmian ilości na aukcjach nieaktywnych ?

pozdrawiamy wwwfactory

**Comment by PawelTaberski** [comment]
@wwwfactory Odnośnie Twoich pytań:
1/ Nie ma możliwości tym zasobem zmiany w x ofertach różnych cen/ilości. Możesz wywołać x razy ten zasób, bądź skorzystać z zasobu PUT /sale/offers/{offerId} (jednakże w nim musisz przesyłać wszystkie pola oferty nawet jak zmieniasz tylko ilość.
2/ Możesz zakończyć przy pomocy zasobu PUT /sale/offer-publication-commands/{commandId} wiele ofert jednym requestem, także byłoby to niepotrzebne dublowanie funkcjonalności.
3/ Ta metoda nie zmienia statusu oferty do tego służy zasób PUT /sale/offer-publication-commands/{commandId}
